src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1610,13 +1624,25 @@ private static Document ParseDocumentPartial(JsonOperationContext context, ref T
             return result;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Document ParseRawDataSectionDocumentWithValidation(JsonOperationContext context, ref TableValueReader tvr, int expectedSize)
         {
             tvr.Read((int)DocumentsTable.Data, out int size);
             if (size > expectedSize || size <= 0)
                 throw new ArgumentException("Data size is invalid, possible corruption when parsing BlittableJsonReaderObject", nameof(size));
 
-            return ParseDocument(context, ref tvr, DocumentFields.All);
+            return new Document
+            {
+                StorageId = tvr.Id,
+                LowerId = TableValueToString(context, (int)DocumentsTable.LowerId, ref tvr),
+                Id = TableValueToId(context, (int)DocumentsTable.Id, ref tvr),
+                Etag = TableValueToEtag((int)DocumentsTable.Etag, ref tvr),
+                Data = new BlittableJsonReaderObject(tvr.Read((int)DocumentsTable.Data, out int dataSize), dataSize, context),
```

Comment:
I guess because the context is of this method `JsonOperationContext` instead of `DocumentsOperationContext`.
If we want to avoid this code duplication I think we need to try and cast the context to `DocumentsOperationContext` inside the `CompressedBlittableJsonReaderObject`

Reply:
this method called only from Voron Recovery tool: https://github.com/ravendb/ravendb/blob/80ba404673f502f2db71269ecba1a135e56e7c95/tools/Voron.Recovery/Recovery.cs#L1495

We don't have `Transaction` there, therefore you don't have compression buffer that we want to dispose, so I returned  `new Document(){}` here explicitly.